src/Exception/InvalidDataException.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types=1);
+
+namespace SupportPal\ApiClient\Exception;
+
+use Throwable;
+
+class InvalidDataException extends InvalidArgumentException
+{
+    /** @var array<mixed> */
+    private $data;
+
+    /**
+     * InvalidDataException constructor.
+     * @param array<mixed> $data
+     * @param string $message
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct(array $data, string $message = '', int $code = 0, Throwable $previous = null)
```

Comment:
TIL: 
> One exception to this rule are arguments of the form Type $param = null, where the null default makes the type implicitly nullable. This usage remains allowed, though it is recommended to use an explicit nullable type instead. 

https://www.php.net/manual/en/functions.arguments.php#example-155

```php
function bar(A $a = null, $b) {} // Still allowed
function bar(?A $a, $b) {}       // Recommended
```

I wonder if there is a `phpcs` rule to prevent this.

Reply:
I will enable https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintsnullabletypefornulldefaultvalue-

Reply:
See https://github.com/supportpal/api-client-php/pull/110

Reply:
will rebase and fix once that is merged